cpp/src/arrow/extension/fixed_shape_tensor_test.cc

```diff
@@ -321,6 +322,70 @@ TEST_F(TestExtensionType, TestFromTensorType) {
   }
 }
 
+void CheckToTensor(const std::vector<int64_t> values, const int32_t cell_size,
+                   const std::vector<int64_t> cell_shape,
+                   const std::vector<int64_t> cell_permutation,
+                   const std::vector<std::string> cell_dim_names,
+                   const std::vector<int64_t> tensor_shape,
+                   const std::vector<std::string> tensor_dim_names,
+                   const std::vector<int64_t> tensor_strides) {
+  auto buffer = Buffer::Wrap(values);
+  const std::shared_ptr<DataType> cell_type = fixed_size_list(int64(), cell_size);
+  std::vector<std::shared_ptr<Buffer>> buffers = {nullptr, buffer};
+  auto arr_data = std::make_shared<ArrayData>(int64(), values.size(), buffers);
+  auto arr = std::make_shared<Int64Array>(arr_data);
+  ASSERT_OK_AND_ASSIGN(auto fsla_arr, FixedSizeListArray::FromArrays(arr, cell_type));
+
+  ASSERT_OK_AND_ASSIGN(
+      auto expected_tensor,
+      Tensor::Make(int64(), buffer, tensor_shape, tensor_strides, tensor_dim_names));
+  const auto ext_type =
+      fixed_shape_tensor(int64(), cell_shape, cell_permutation, cell_dim_names);
+
+  auto ext_arr = ExtensionType::WrapArray(ext_type, fsla_arr);
+  const auto tensor_array = std::static_pointer_cast<FixedShapeTensorArray>(ext_arr);
+  ASSERT_OK_AND_ASSIGN(const auto actual_tensor, tensor_array->ToTensor());
```

Comment:
Why is it called `ToTensor` here but `MakeTensor` for scalars?

Reply:
Changed.